githubmarkdownui/ci.py

```diff
@@ -40,6 +44,38 @@ class CITask:
     executed within a build pipeline."""
     metadata: CITaskMetadata
 
+    def get_left_justified_task_string_length(self) -> int:
+        """Calculates the length of the task string that will be left justified in the task list. This consists of the task
+        emoji, task name, and task duration.
+        """
+        # Factor emoji, 2 space buffer, task name, one space, left bracket, task duration, right bracket in string length.
+        return len(f'{self.metadata.status.value}  {self.metadata.name}'
+                   f'{" (" + self.metadata.duration + ")" if self.metadata.duration else ""}')
+
+    def build_task_string(self, width: int = 0) -> str:
+        """Builds the string to represent the given task in a task list. The string will be formatted as follows:
+
+        <emoji>  <task name> (<task duration>)   <task info>
+
+        The task duration will be bolded, and the emoji, task name, and task duration will be left justified according to
+        the given width.
+
+        :param width: The total width of the string, used to left justify the contents
+        """
+        duration_string = ' ' + bold(f'({self.metadata.duration})') if self.metadata.duration else ''
+        # Since the strong tags are part of the duration string, they will be accounted for when left justifying with the
+        # given width. But when this string is used in GitHub Flavored Markdown, the strong tags will not show up since they
+        # will bold the text inside it, so we must add additional padding to the width according to the length of the strong
+        # tags. The only string that should not have additional padding is the longest string, since that will just increase
+        # the distance between the duration and the additional info.
+        additional_padding = len(bold('')) if self.metadata.duration and \
+            self.get_left_justified_task_string_length() != width else 0
+
+        left_justified_string = f'{self.metadata.status.value}  {self.metadata.name}{duration_string}'
+        info_string = '   ' + self.metadata.info if self.metadata.info else ''
```

Comment:
To confirm, in Mike's example, tasks could have additional links to log/build scan, do we expect `self.metadata.info` is already in hyperlinked markdown syntax?

Reply:
Yeah, I wanted to keep `self.metadata.info` general enough for people to put whatever they want. We can pass in something like `[link('logs', <log url>)]`